src-manual/Vulkan/Requirements.hs

```diff
@@ -184,162 +237,246 @@ data Requirements where
   Requirements
     :: forall setts feats props
     .  ( KnownChain setts, Extendss InstanceCreateInfo        setts
-       , KnownChain feats, Extendss PhysicalDeviceFeatures2   feats
+       , KnownChain feats, Extendss PhysicalDeviceFeatures2   feats, Extendss DeviceCreateInfo feats
        , KnownChain props, Extendss PhysicalDeviceProperties2 props
        )
     => {
-       -- | Minimum Vulkan API version required.
-         version            :: Word32 
-       -- | Required Vulkan layer names and their minimum versions.
-       , instanceLayers     :: HashMap ByteString Word32
-       -- | Required Vulkan instance extension names and their minimum versions,
+       -- | Required and optional minimum Vulkan API version.
+         version                 :: ReqAndOpt Word32
+       -- | Required and optional Vulkan layer names and their minimum versions.
+       , instanceLayers          :: ReqAndOpt (HashMap ByteString Word32)
```

Comment:
OK, I didn't know that. Are there some layers that need to be enabled before others then?

Reply:
not necessarily, but consider where `api_dump` goes for example, you might want it first in the list (next to the application) to avoid capturing all the vulkan commands generated by subsequent layers.

Reply:
Makes sense yeah.

Reply:
Done.